src/resmom/stage_func.c

```diff
@@ -1959,36 +1988,54 @@ sys_copy(int dir, int rmtflg, char *owner, char *src, struct rqfpair *pair, int
 			if ((pw->pw_userlogin == INVALID_HANDLE_VALUE) && (strcmpi(owner, getlogin()) == 0)) {
 				sprintf(log_buffer, "CreateProcess(%s) under acct %s wdir=%s",
 					cmd_line, owner, wdir);
-				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, "sys_copy", log_buffer);
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, __func__, log_buffer);
 				rc = CreateProcess(NULL, cmd_line,
 					NULL, NULL, TRUE, flags,
 					NULL, wdir, &si, &pi);
+				if (rc == 0) {
+					errno = GetLastError();
+					snprintf(log_buffer, sizeof(log_buffer), "CreateProcess failed with status=%d", rc);
+					log_err(errno, __func__, log_buffer);
+				}
 			} else {
 				sprintf(log_buffer, "CreateProcessAsUser(%d, %s) under acct %s wdir=%s",
 					pw->pw_userlogin, cmd_line, getlogin(), wdir);
-				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, "sys_copy", log_buffer);
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, __func__, log_buffer);
 				rc = CreateProcessAsUser(pw->pw_userlogin, NULL, cmd_line,
 					NULL, NULL, TRUE, flags,
 					NULL, wdir, &si, &pi);
+				if (rc == 0) {
+					errno = GetLastError();
+					snprintf(log_buffer, sizeof(log_buffer), "CreateProcessAsUser failed with status=%d", rc);
+					log_err(errno, __func__, log_buffer);
+				}
 			}
 
-			if (rc == 0)
-				errno = GetLastError();
-
 			close(fd);
 			fd = -1; /* already done the close */
 
 			if (errno != 0) {
 				rc = errno + 10000;	/* error on fork (100xx), retry */
 			} else {
 				if (WaitForSingleObject(pi.hProcess,
-					INFINITE) != WAIT_OBJECT_0)
-					errno = GetLastError();
-				else if (!GetExitCodeProcess(pi.hProcess, &rc))
-					errno = GetLastError();
+					INFINITE) != WAIT_OBJECT_0) {
+					snprintf(log_buffer, sizeof(log_buffer), " WaitForSingleObject failed with status=%d", rc);
+					log_err(-1, __func__, log_buffer);
+				}
+				else if (!GetExitCodeProcess(pi.hProcess, &rc)) {
+					snprintf(log_buffer, sizeof(log_buffer), "GetExitCodeProcess failed with status=%d", rc);
+					log_err(-1, __func__, log_buffer);
+				}
 
-				CloseHandle(pi.hProcess);
-				CloseHandle(pi.hThread);
+				if (!CloseHandle(pi.hProcess)) {
```

Comment:
You don't need handle error case for CloseHandle. It will be taken care as part of this PR by Shrini #1727 
Same suggestion for line number 2035, 2065 and 2072